tests/common/test_additional_secrets_redaction.py

```diff
@@ -0,0 +1,402 @@
+        self,
+        call_racf_mock: Mock,
+    ):
+        secret_trait = "base:universal_access"
+        data_set_admin = DataSetAdmin(additional_secret_traits=[secret_trait])
+        call_racf_mock.side_effect = [
+            TestDataSetConstants.TEST_EXTRACT_DATA_SET_RESULT_BASE_ONLY_SUCCESS_XML,
+            TestDataSetConstants.TEST_ALTER_DATA_SET_RESULT_ERROR_XML,
+        ]
+        with self.assertRaises(SecurityRequestError) as exception:
+            data_set_admin.alter(
+                "ESWIFT.TEST.T1136242.P3020470",
+                traits=TestDataSetConstants.TEST_ALTER_DATA_SET_REQUEST_TRAITS,
+            )
+        self.assertEqual(
+            exception.exception.result,
+            TestCommonConstants.TEST_ALTER_DATA_SET_RESULT_ERROR_UACC_SECRET_DICTIONARY,
+        )
+        self.assertNotIn(
+            TestDataSetConstants.TEST_ALTER_DATA_SET_REQUEST_TRAITS[f"{secret_trait}"],
+            str(exception.exception.result),
+        )
+
+    def test_data_set_admin_custom_secret_redacted_request(self):
+        data_set_admin = DataSetAdmin(
+            additional_secret_traits=["base:owner"], generate_requests_only=True
+        )
+        result = data_set_admin.alter(
+            "ESWIFT.TEST.T1136242.P3020470",
+            traits=TestDataSetConstants.TEST_ALTER_DATA_SET_REQUEST_TRAITS,
+        )
+        self.assertNotIn(
+            TestDataSetConstants.TEST_ALTER_DATA_SET_REQUEST_TRAITS["base:owner"],
+            str(result),
+        )
+
+
+@patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+class TestAdditionalSecretsLoggingRedaction(unittest.TestCase):
+    maxDiff = None
+    user_admin = UserAdmin(debug=True)
+    ansi_escape = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
+
+    # ============================================================================
+    # Add Additional Secrets
+    # ============================================================================
+    def test_alter_user_request_debug_log_additional_secret_added_get_redacted_on_success(
+        self,
+        call_racf_mock: Mock,
+    ):
+        user_admin = UserAdmin(debug=True, additional_secret_traits=["omvs:uid"])
+        call_racf_mock.side_effect = [
+            TestUserConstants.TEST_EXTRACT_USER_RESULT_BASE_ONLY_SUCCESS_XML,
+            TestUserConstants.TEST_ALTER_USER_RESULT_EXTENDED_SUCCESS_XML,
+        ]
+        stdout = io.StringIO()
+        with contextlib.redirect_stdout(stdout):
+            user_admin.alter(
+                "squidwrd",
+                traits=TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_EXTENDED,
+            )
+        success_log = self.ansi_escape.sub("", stdout.getvalue())
+        self.assertEqual(
+            success_log,
+            TestCommonConstants.TEST_ALTER_USER_ADDITIONAL_SECRET_ADDED_SUCCESS_LOG,
+        )
+        self.assertNotIn(
+            TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_EXTENDED["omvs:uid"],
+            success_log,
```

Comment:
`TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_EXTENDED["omvs:uid"]` should be casted to a string here, no?

Reply:
good catch, resolved.